### PR TITLE
[Relax] Integrate cuDNN attention

### PR DESCRIPTION
**Pull request by vinx13**
This integrates cuDNN attention kernels to BYOC.
A dependency of cudnn_frontend is added.
The cuDNN attention kernel supports fused qkv in BS3NH and SBN3H layouts.

cc @sunggg @masahi @yongwww @tqchen

### CODE REVIEW

**Review by yongwww — APPROVED** [review]

File: python/tvm/contrib/cutlass/gen_tensor_op.py [hunk 1]
```diff
@@ -745,8 +745,8 @@ def get_batch_on_arg(arg_name, arg_shape):
             attrs["qkv"] = func_args[0]
             attrs["num_queries"] = s = annotations["num_queries"]
             attrs["num_keys"] = annotations["num_keys"]
-            if len(func_args) > 5 and not is_var_len:  # +1 for workspace, the last arg
-                attrs["bias"] = func_args[4]
+            if len(func_args) > 2 and not is_var_len:  # +1 for workspace, the last arg
```
Comment (sunggg) [inline python/tvm/contrib/cutlass/gen_tensor_op.py:748]: I guess cutlass side is mostly the interface change? 
Or is there any functional change?

**Review by sunggg — COMMENTED** [review]
Thank you @vinx13 for the new addition! 
Overall, looks good to me.
Would you describe high-level strategy for attention somewhere? (e.g., when to offload cudnn, cutlass, TIR, etc.)
If this is about landing machinery rather than such offloading decision, would appreciate if you can provide some recommendations.

**Comment by vinx13** [comment]
The new attention can be applied via cudnn BYOC. The decision of which BYOC backend (cudnn, cutlass) to use is left to the users. cudnn is likely to perform better on H100 as it has specific optimizations

File: python/tvm/contrib/cutlass/gen_tensor_op.py (hunk 1 above)
Comment (vinx13) [reply python/tvm/contrib/cutlass/gen_tensor_op.py:748]: FuseOps is updated to inline the constant shape parameters. So there are fewer func_args here

**Review by masahi — APPROVED** [review]
I remember cuDNN attention supports fp8, would be interesting to support that too.

**Pull request merged**